blog/content/post/2024-01-31-drm-bridge-chaining.md

```diff
@@ -0,0 +1,863 @@
+    +                       return ret;
+    +               }
+
+    +               drm_connector_helper_add(&it6263->connector,
+    +                                        &it6263_connector_helper_funcs);
+    +               drm_connector_attach_encoder(&it6263->connector, bridge->encoder);
+    +       }
+            return ret;
+     }
+
+    @@ -794,6 +790,42 @@ static u32
+            return input_fmts;
+     }
+
+    +static enum drm_connector_status
+    +it6263_bridge_detect(struct drm_bridge *bridge)
+    +{
+    +       struct it6263 *it6263 = bridge_to_it6263(bridge);
+    +
+    +       if (it6263_hpd_is_connected(it6263))
+    +               return connector_status_connected;
+    +
+    +       return connector_status_disconnected;
+    +}
+    +
+    +static int it6263_bridge_get_modes(struct drm_bridge *bridge, struct drm_connector *connector)
+    +{
+    +       struct it6263 *it6263 = bridge_to_it6263(bridge);
+    +       u32 bus_format = MEDIA_BUS_FMT_RGB888_1X24;
+    +       struct edid *edid;
+    +       int num = 0;
+    +       int ret;
+    +
+    +       edid = drm_do_get_edid(connector, it6263_read_edid, it6263);
+    +       drm_connector_update_edid_property(connector, edid);
+    +       if (edid) {
+    +               num = drm_add_edid_modes(connector, edid);
+    +               it6263->is_hdmi = drm_detect_hdmi_monitor(edid);
+    +               kfree(edid);
+    +       }
+    +
+    +       ret = drm_display_info_set_bus_formats(&connector->display_info,
+    +                                              &bus_format, 1);
+    +       if (ret)
+    +               dev_dbg(&it6263->hdmi_i2c->dev,
+    +                       "failed to set the supported bus format %d\n", ret);
+    +
+    +       return num;
+    +}
+    +
+     static const struct drm_bridge_funcs it6263_bridge_funcs = {
+            .attach = it6263_bridge_attach,
+            .atomic_duplicate_state = drm_atomic_helper_bridge_duplicate_state,
+    @@ -804,6 +836,11 @@ static const struct drm_bridge_funcs it6263_bridge_funcs = {
+            .enable = it6263_bridge_enable,
+            .atomic_check = it6263_bridge_atomic_check,
+            .atomic_get_input_bus_fmts = it6263_bridge_atomic_get_input_bus_fmts,
+    +
+    +       /*We need to export these functions in case we create external
+    +        *drm_connector.*/
+    +       .detect = it6263_bridge_detect,
+    +       .get_modes = it6263_bridge_get_modes,
+     };
+
+     static int it6263_check_chipid(struct it6263 *it6263)
+    @@ -991,6 +1028,8 @@ static int it6263_probe(struct i2c_client *client,
+
+            it6263->bridge.funcs = &it6263_bridge_funcs;
+            it6263->bridge.of_node = np;
+    +       it6263->bridge.type = DRM_MODE_CONNECTOR_HDMIA;
+    +       it6263->bridge.ops = DRM_BRIDGE_OP_DETECT | DRM_BRIDGE_OP_MODES;
+            drm_bridge_add(&it6263->bridge);
+
+            i2c_set_clientdata(client, it6263);
+    ```
```

Comment:
There are too many code snippets here. We provide the whole solution in an incorrect way. A repository would be better, but we don't want to go that path. The only way I see it may work is referencing some other code and saying "do this and that in similar way as here<link to some code>"

Reply:
When I was digging with this problem I could not find many examples, so, I doubt `referencing some other code and saying "do this and that in similar way as here"` is a solution here. On the other hand this snippets could be deleted, so only list of steps performed stays. Here 30a3d98b6b3b9e0e566ee29105df35aeca561bc1 is a commit which deletes the snippets.